diff --git a/lib/common/index.js b/lib/common/index.js
index ace2c1a19674ff71a418daee920fbf5f26d6a8d4..fa5b9a9f1812a04e12ad6497ad163e121945bec1 100644
--- a/lib/common/index.js
+++ b/lib/common/index.js
@@ -869,7 +869,7 @@ function resolve(specifier, context, nextResolve) {
     const filename = import_url.default.fileURLToPath(context.parentURL);
     const resolved = resolveHook(filename, specifier);
     if (resolved !== void 0) {
-      specifier = context.conditions?.includes("import") ? import_url.default.pathToFileURL(resolved).toString() : resolved;
+      specifier = [...(context.conditions ?? [])].includes("import") ? import_url.default.pathToFileURL(resolved).toString() : resolved;
     }
   }
   const result2 = nextResolve(specifier, context);
diff --git a/lib/transform/esmLoader.js b/lib/transform/esmLoader.js
index 2ae509ca7a981bb75c9cc5835fbe252fad231881..9144006dad7c3e9f634b72b75e05978848254d3e 100644
--- a/lib/transform/esmLoader.js
+++ b/lib/transform/esmLoader.js
@@ -5629,7 +5629,7 @@ function resolve(specifier, context, nextResolve) {
     const filename = import_url.default.fileURLToPath(context.parentURL);
     const resolved = resolveHook(filename, specifier);
     if (resolved !== void 0) {
-      specifier = context.conditions?.includes("import") ? import_url.default.pathToFileURL(resolved).toString() : resolved;
+      specifier = [...(context.conditions ?? [])].includes("import") ? import_url.default.pathToFileURL(resolved).toString() : resolved;
     }
   }
   const result2 = nextResolve(specifier, context);
